sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -767,6 +777,35 @@ case class InSet(child: Expression, hset: Set[Any]) extends UnaryExpression with
   override protected def withNewChildInternal(newChild: Expression): InSet = copy(child = newChild)
 }
 
+object InSet {
+  class LCaseSet(inputSet: Set[Any]) extends immutable.Set[Any] with Serializable {
+    private val strSet = inputSet.map { s =>
+      if (s == null) null
+      else CollationAwareUTF8String.lowerCaseCodePoints(s.asInstanceOf[UTF8String])
+    }
+    override def incl(elem: Any): Set[Any] = inputSet.incl(elem)
```

Comment:
shouldn't these methods also return a `LCaseSet` (same  for `excl` and both methods in the other class)?

Reply:
The methods shouldn't be invoked at all. Let me raise an error here.